File: R-KNN.R
library(class) # Package required for KNN
library(caret) # Package required for confusion matrix

# Importing data
mnist_raw <-
  read.csv(file.path("C:\\Users\\esbe1\\Desktop\\Portfolio", "mnist_train.csv"))

# Required inputs
size <- 2000 # Total observations included
split <- 50  # Split size where 90 means that it is 90/10
k <- 1

start <- Sys.time()

# Splitting between train and test data
mnist_train <- mnist_raw[1:(size * split * 0.01),-1]
mnist_test <- mnist_raw[(size * split * 0.01 + 1):size,-1]

# Splitting between train and test labels
mnist_train_labels <-
  factor(mnist_raw[1:(size * split * 0.01), 1])
mnist_test_labels <-
  factor(mnist_raw[(size * split * 0.01 + 1):size, 1])

# Training the algorithm
mnist_pred <-
  knn(
    train = mnist_train ,
    test = mnist_test ,
    cl = mnist_train_labels,
    k = k
  )

# Confusion Matrix and calculation of overall accuracy
cf_m  <- confusionMatrix(mnist_test_labels , mnist_pred)
cf_table <- cf_m$table
acc <- sum(diag(cf_table)) / sum(cf_table)

#logging results

stop <- Sys.time()
time <- stop - start
write.table(cf_table, "C:/Users/esbe1/Desktop/Portfolio/R-KNN-CF.txt", sep = ",")

File: R-KNN.py
# -*- coding: utf-8 -*-
"""
Created on Wed Oct 14 12:55:03 2020

@author: Esben Gammelgaard
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


observations = 2000
test_size = 0.5
k = 1

#%% Importing data
path = "C:\\Users\\esbe1\\Desktop\\Portfolio\\mnist_train.csv"

# Read dataset to pandas dataframe
dataset = pd.read_csv(path)
dataset = dataset.loc[:observations-1 ,:]
#%% splitting for train and test 
from sklearn.model_selection import train_test_split

mnist = dataset.iloc[:, :-1].values
mnist_labels = dataset.iloc[:, 0].values

mnist_train, mnist_test, mnist_train_labels,mnist_test_labels = train_test_split(mnist, mnist_labels, test_size=test_size)

#%% Training the predictor

from sklearn.neighbors import KNeighborsClassifier
classifier = KNeighborsClassifier(n_neighbors=k)
classifier.fit(mnist_train, mnist_train_labels)

#%% Predicting the values of test data 

mnist_pred = classifier.predict(mnist_test) 

#%% Evaluating the performance

from sklearn.metrics import classification_report, confusion_matrix
print(confusion_matrix(mnist_test_labels, mnist_pred))
print(classification_report(mnist_test_labels, mnist_pred))

